contracts/libs/0.8.x/MinterUtils_v0_1_1.sol

```diff
@@ -0,0 +1,276 @@
+                .getPrimaryRevenueSplits(projectId, pricePerTokenInWei);
+        } else {
+            // get engine splits
+            uint256 platformProviderRevenue_;
+            address payable platformProviderAddress_;
+            (
+                renderProviderRevenue_,
+                renderProviderAddress_,
+                platformProviderRevenue_,
+                platformProviderAddress_,
+                artistRevenue_,
+                artistAddress_,
+                additionalPayeePrimaryRevenue_,
+                additionalPayeePrimaryAddress_
+            ) = IGenArt721CoreContractV3_Engine(genArtCoreContract)
+                .getPrimaryRevenueSplits(projectId, pricePerTokenInWei);
+            // Platform Provider payment (only possible if engine)
+            if (platformProviderRevenue_ > 0) {
+                _projectCurrency.transferFrom(
+                    msg.sender,
+                    platformProviderAddress_,
+                    platformProviderRevenue_
+                );
+            }
+        }
+        // Art Blocks payment
+        if (renderProviderRevenue_ > 0) {
+            _projectCurrency.transferFrom(
+                msg.sender,
+                renderProviderAddress_,
+                renderProviderRevenue_
+            );
+        }
+        // artist payment
+        if (artistRevenue_ > 0) {
+            _projectCurrency.transferFrom(
+                msg.sender,
+                artistAddress_,
+                artistRevenue_
+            );
+        }
+        // additional payee payment
+        if (additionalPayeePrimaryRevenue_ > 0) {
+            _projectCurrency.transferFrom(
+                msg.sender,
+                additionalPayeePrimaryAddress_,
+                additionalPayeePrimaryRevenue_
+            );
+        }
+    }
+
+    function getV3CoreIsEngine(
+        IGenArt721CoreContractV3_Base genArt721CoreV3_Base
+    ) internal returns (bool isEngine) {
+        // determine if core is engine based off of coreType() response
+        isEngine =
+            keccak256(abi.encodePacked(genArt721CoreV3_Base.coreType())) ==
+            keccak256(abi.encodePacked(FLAGSHIP_CORE_TYPE));
+        // validate the above logic by confirming the payment split response
+        _validateV3CoreGetPrimaryRevenueSplitsResponse(
+            isEngine,
+            address(genArt721CoreV3_Base)
+        );
+        return isEngine;
+    }
+
+    /**
+     * @notice Validates that a GenArt721CoreV3 core contract's
+     * `getPrimaryRevenueSplits` function returns the expected number of
+     * return values based on the `isEngine` expected state of the core
+     * contract.
+     * @param isEngine Whether the core contract is expected to be an Art
+     * Blocks Engine contract or not.
+     * @param genArt721CoreV3 The address of the deployed core contract.
+     */
+    function _validateV3CoreGetPrimaryRevenueSplitsResponse(
+        bool isEngine,
```

Comment:
Should this be passed in, or should this just be determined internally (via a call to `getV3CoreIsEngine`)?

Reply:
Ah wait–I see; the abstraction is happening this way, it is just happening in reverse, with `getV3CoreIsEngine` being the `internal` entry point and this method being `private`.

Reply:
Yeah, thought we could let the library simply get and return if the core is engine, while validating that answer internally.

Reply:
Yep yep–makes sense; resolving.